operator/v1/types_ingress.go

```diff
@@ -1399,6 +1399,138 @@ type IngressControllerHTTPHeaders struct {
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be modified for TLS passthrough
+	// connections.
+	// Setting the HSTS (`Strict-Transport-Security`) header is not supported via actions. `Strict-Transport-Security `
+	// may only be configured using the "haproxy.router.openshift.io/hsts_header" route annotation, and only in
+	// accordance with the policy specified in Ingress.Spec.RequiredHSTSPolicies.
+	// Any actions defined here are applied after any actions related to the following other fields:
+	// cache-control, spec.clientTLS,
+	// spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId,
+	// and spec.httpHeaders.headerNameCaseAdjustments.
+	// Header actions defined for request here will not take precedence over any header modifying action or route annotation defined on Route objects.
+	// Header actions defined for response here will take precedence over any header modifying action or route annotation defined on Route objects.
+	// Setting httpCaptureHeaders for the headers set via this API will not work as the headers set via this API will not appear in access logs.
+	// Setting "cookie" or "set-cookie" headers that conflict with OpenShift router's session cookies will result in undefined behavior.
+	// Note that the total size of all net added headers *after* interpolating dynamic values
+	// must not exceed the value of spec.tuningOptions.headerBufferMaxRewriteBytes on the
+	// IngressController. Please refer https://github.com/openshift/api/blob/61a5301895a5/operator/v1/types_ingress.go#L1422 for `HeaderBufferBytes`
+	// and https://github.com/openshift/api/blob/61a5301895a5/operator/v1/types_ingress.go#L1442 for `HeaderBufferMaxRewriteBytes` for more details.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActions defines configuration for actions on HTTP request and response headers.
+type IngressControllerHTTPHeaderActions struct {
+	// response is a list of HTTP response headers to modify.
+	// The cluster administrator uses these to specify HTTP response headers that should be set or
+	// deleted at the global level when forwarding responses from an application to the client, i.e. for all connections handled by the ingress controller.
+	// Currently actions may define to either `Set` or `Delete` headers values.
+	// Actions are applied in sequence as defined in this list.
+	// A maximum of 20 response header actions may be configured.
+	// Sample fetchers allowed are "res.hdr" and "ssl_c_der".
+	// Converters allowed are "lower" and "base64".
+	// Example header values: "%[res.hdr(X-target),lower]", "%{+Q}[ssl_c_der,base64]".
+	// + ---
+	// + Note: Any change to regex mentioned below must be reflected in the CRD validation of route in https://github.com/openshift/library-go/blob/master/pkg/route/validation/validation.go and vice-versa.
```

Comment:
This note is applicable to the Route API; is it applicable to the IngressController API?

Reply:
I think to maintain consistency it will be better to keep this here as well.

Reply:
Why would you keep an untrue statement for "consistency"?

Reply:
anyway I am ok to remove this.

Reply:
All right, I won't insist on changing this.  I don't mind too much if you keep the comment as is.  You are right that it does make sense to _look at_ the Route validation when updating the IngressController validation.  The Route and IngressController validation do have differences (namely that IngressController forbids modifying the `host` header whereas the Route API allows modifying it), and this regexp could potentially deviate in the future, but right now, the regexps are the same, and we can always change the comment in the future if we find a reason to change the regexps to be different.

Reply:
I think I will remove it as it will be better to keep them seperate for IC and route.